### DISCUSSION TITLE
Would you ever make an Impact Framework plugin?

### DISCUSSION THREAD

**Discussion by rachaelcodes** [mlco2/ecologits#94] [Ideas]
Hi 👋  Firstly I just wanted to say thanks for putting this project together - it's exactly what I've been looking for in terms of estimating the carbon impact of LLM calls.

I think this tool could underpin a really good [plugin](https://explorer.if.greensoftware.foundation/) for the Green Software Foundation's open source [Impact Framework](https://if.greensoftware.foundation/). Is this something your team have considered? Or would you be open to collaboration? 

I'm not affiliated with the GSF -  I've just been experimenting with the Impact Framework for aggregating data about carbon emissions across website infrastructure. There's definitely some room for tools that account for the impact of 3rd party LLM requests.

**Comment by samuelrince** [comment]
Hi @rachaelcodes,

Thanks a lot for the feedback! 🙏 

Indeed, I've heard about the Impact Framework and the SCI defined by the GSF, but we haven't worked on it for now. To be honest, I am not familiar with how it works and how we could be integrated as a plugin, but definitely eager to learn more about it!

Do you have some resources on the subject, or are you willing to help us on it? We can schedule a meeting if that's easier! 🤗

**Reply by rachaelcodes** [reply]
Thanks for getting back to me so quickly! A meeting would be great - what's the best way to contact you to arrange it? I'd be very happy to get involved with building this.

There's an introductory video of the impact framework from GSF [here](https://youtu.be/msk-55owTeM) and a (quite long) blogpost on using it to measure a website [here](https://medium.com/digital-futures-told/how-to-use-the-impact-framework-to-measure-the-environmental-impact-of-your-website-c66bc9b03f9b) . The 'plugins' are JS packages that take in user defined input (e.g. token use and model choice) and output a carbon and/or other impact estimate - a lot like your calculator on Hugging Face. The idea of the Impact Framework is to allow users to chain a lot of these plugins together and come up with some grand calculation for the total impact of their service.

There's an example plugin here: https://github.com/jmcook1186/carbon-intensity-plugin .

**Reply by samuelrince** [reply]
I've just sent you an email (taken from your personal website)!
